[Brandon] iP

brelkh opened the pull request:
### Mike

> “The bad news is time flies. The good news is you’re the pilot”

Mike is a *simple task manager* bot. It is
- Easy to learn and use
- Fast (if you can type fast)
- FREE 💵 

It can:
1. Add tasks (3 types - todos, deadlines, events) to a list
2. Perform basic functions on the list
3. Store the list in hard drive for future reference.

To use it,
- [x] Download the jar file
- [ ] Run the jar file with java
- [ ] Add your tasks!

Know [Java](https://docs.oracle.com/en/java/)? Feel free to fork it and edit code as you please!
```java
public class Main {
    public static void main(String[] args) {
        
    }
}
```

Commit:
Add Gradle support

brelkh committed:
add level 1 componenets

brelkh committed:
add Mike.java; edit level 1 components

brelkh committed:
add level 2 components (add, list)

brelkh committed:
add task.java and level 3 components

brelkh committed:
add level 4 components (todo, event, deadline)

brelkh committed:
add automated text UI testing capabilities

brelkh committed:
add error handling capabilities

brelkh committed:
add delete (remove) functionality

brelkh committed:
edit list method

brelkh committed:
Edit switch statement indentation

brelkh committed:
add ExistInSeven.java branch test file

brelkh committed:
Add level 7 components (save feature).

brelkh committed:
Import java.time libraries.

brelkh committed:
Update gitignore file to include storedList.txt

brelkh committed:
Merge branch 'branch-Level-7'

brelkh committed:
Merge branch 'branch-Level-8'

brelkh committed:
Add level 8 components (replace Strings with DateTimes).

brelkh committed:
Add "More-OOP" iP increment.

brelkh committed:
Edit files for A-Packages component (organize files into packages)

brelkh committed:
Rearrange files into packages more neatly.

brelkh committed:
Add A-JUnit components (junit test files).

brelkh committed:
Add Jar file for A-Jar component

brelkh committed:
Remove some completed "todo" comments.

brelkh committed:
Update JavaDocs of most public methods.

brelkh committed:
Edit code to adhere to given coding standard (A-CodingStandard).

brelkh committed:
Add Level-9 components (find functionality).

brelkh committed:
Merge branch 'branch-A-JavaDoc'

brelkh committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/mike/Parser.java
#	src/main/java/mike/Storage.java

Resolution:
Remove duplicate line of code

brelkh committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/mike/TaskList.java

brelkh committed:
Edit and tidy Level-9 (find) methods.

jamesyeap commented on src/main/java/mike/Deadline.java:
```diff
@@ -0,0 +1,76 @@
+ */
+public class Deadline extends Task {
+    private final String endDate;
+    private final LocalDate date;
```
Maybe can change the naming of the fields to be more explicit? Probably something like
```java
private final String endDateString;
private final LocalDate endDate;
```

jamesyeap reviewed (commented):
Looks awesome! 😃

YiHern-Lee commented on src/main/java/mike/Parser.java:
```diff
@@ -0,0 +1,137 @@
+
+public class Parser {
+    private final String userInput;
+    private final String[] splitString;
```
Perhaps you can rename `splitString` to `splitStrings`, as it is an array of strings.

YiHern-Lee commented on src/main/java/mike/Mike.java:
```diff
@@ -0,0 +1,238 @@
+                                parser.getUserInput());
+                throw new UnsupportedOperationException(invalidCommandMessage);
+            }
+        } catch(UnsupportedOperationException e) {
```
Maybe you can add whitespaces after your **catch** to be in accordance with the [Java Coding Standards](https://se-education.org/guides/conventions/java/intermediate.html).

YiHern-Lee reviewed (commented):
Overall, your code is very neat and there's minimal coding standards violation. 👍

MelvinCZJ commented on src/main/java/mike/Deadline.java:
```diff
@@ -0,0 +1,76 @@
+        return LocalDate.parse(endDate);
+    }
+
+    public Deadline(String name, String endDate) {
```
This isn't a coding standard violation is an overloaded constructor necessary here❓

MelvinCZJ commented on src/main/java/mike/Deadline.java:
```diff
@@ -0,0 +1,76 @@
+        if (super.isDone) {
+            doneIndicator = "true";
+        }
+        String storedListFormat = String.format("%s|%s|%s|%s", taskType,
```
I love the way you store the data of the tasks here! 😃

MelvinCZJ commented on src/main/java/mike/Deadline.java:
```diff
@@ -0,0 +1,76 @@
+        } else {
+            doneMark = " ";
+        }
+        String dateOutput = this.date.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
```
I wish I knew of the DateTimeFormatter instead of creating my own DateManager and TimeManager classes 😞

MelvinCZJ commented on src/main/java/mike/Event.java:
```diff
@@ -0,0 +1,70 @@
+     *
+     * @return Event object with isDone set to true.
+     */
+    public Event markAsDone() {
```
I like how you returned a new task object instead of modifying the isDone property of the existing task object!

MelvinCZJ commented on src/main/java/mike/Event.java:
```diff
@@ -0,0 +1,70 @@
+     *
+     * @return Event object with isDone set to false.
+     */
+    public Event markAsUndone() {
```
Is there a way to check whether a task is marked/unmarked before letting users unmark/mark a task? I think it might be better if the user knows that they are trying to mark an already marked task or trying to unmark an unmarked task. (Just a suggestion! 😄)

MelvinCZJ commented on src/main/java/mike/Mike.java:
```diff
@@ -0,0 +1,238 @@
+    public void run() {
+        ui.printGreeting();
+        ui.printStartingInstruction();
+        loadStoredList();
```
Nothing wrong with this, but maybe this part could be delegated to the Storage class? You could have a load method in Storage that returns a new arraylist created from the saved tasklist instead of having your main class handle the loading. (Just a suggestion! Please don't take any offense!)

MelvinCZJ commented on src/main/java/mike/Parser.java:
```diff
@@ -0,0 +1,137 @@
+     *
+     * @return boolean representing whether the user has input the exit command.
+     */
+    public boolean isBye() {
```
Clear naming! 👍

MelvinCZJ reviewed (commented):
Generally clean code that is easy to understand with little coding style error for me to commend on. (Since I don't really want to point out the same things that others have already pointed out)

brelkh committed:
Merge remote-tracking branch 'origin/add-gradle-support'

brelkh committed:
Add A-Gradle components

brelkh committed:
Add A-Checkstyle components

brelkh committed:
Edit README.md

brelkh committed:
Add temporary Duke (GUI) files alongside functioning Mike.

brelkh committed:
Add Level 10 (GUI) components

brelkh committed:
Add assert statements to important assumptions.

Currently, the program lacks any Assert statements. However, it is
important for programmers not to assume the code will always work
as intended.

This commit adds assert statements to document important assumptions.
Java assert statements have been used so that the failed assertions will
be detected when the program is run with Gradle.

Note: be sure to enable assertions in the build.gradle file to verify
assertions

brelkh committed:
Add another important assert statement.

brelkh committed:
Update certain outdated javadoc comments.

brelkh committed:
Remove unused methods to improve code quality (A-CodeQuality component)

brelkh committed:
Merge branch 'master' into branch-A-Assertions

brelkh committed:
Merge pull request #3 from brelkh/branch-A-CodeQuality

Remove unused methods to improve code quality (A-CodeQuality component)

brelkh committed:
Merge branch 'branch-A-Assertions'

# Conflicts:
#	src/main/java/mike/Mike.java

brelkh committed:
Allow shorter aliases for list, todo, deadline and event keywords.

brelkh committed:
Merge pull request #2 from brelkh/branch-A-Assertions

Branch a assertions

brelkh committed:
Use smaller, circular profile pictures in app rather than squares.

brelkh committed:
Add credit for JavaFX code from SE-EDU JavaFX tutorial.

brelkh committed:
Remove DaDuke and DaUser placeholder images. (in branch-level-10)

brelkh committed:
Merge branch 'branch-Level-10'

brelkh committed:
Add Ui.png to the docs folder

brelkh committed:
Add User Guide for Mike in README.md

brelkh committed:
Update and touch up code for release

brelkh committed:
Set theme jekyll-theme-cayman

brelkh committed:
Set theme jekyll-theme-slate

brelkh committed:
Merge branch 'master' of https://github.com/brelkh/ip

* 'master' of https://github.com/brelkh/ip:
  Set theme jekyll-theme-slate
  Set theme jekyll-theme-cayman

brelkh committed:
Update README.md
